pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -322,22 +361,84 @@ func (c *podIPMetricsCollector) setupDirectoryWatcher(dir string) error {
 	return nil
 }
 
+func (c *podIPMetricsCollector) countTotalIPs(subnet *net.IPNet) (uint64, error) {
+	ones, bits := subnet.Mask.Size()
+	if bits == 0 {
+		return 0, fmt.Errorf("invalid IP subnet mask %v", subnet.Mask)
+	}
+	if ones == bits {
+		ones = bits - 1
+	}
+	return uint64(1 << (bits - ones)), nil
+}
+
+func (c *podIPMetricsCollector) calculateAssignedIP(subnetFile string) error {
+	var assignedIPv4, assignedIPv6 uint64
+	f, err := os.Open(subnetFile)
+	if err != nil {
+		glog.Errorf("Error opening JSON file %s: %v", subnetFile, err)
+		return err
+	}
+	defer f.Close()
+
+	var config IPAMConfig
+	err = json.NewDecoder(f).Decode(&config)
+	if err != nil {
+		glog.Errorf("Error decoding JSON file %s: %v", subnetFile, err)
+		return err
+	}
+
+	for _, plugin := range config.Plugins {
+		for _, ranges := range plugin.IPAM.Ranges {
+			for _, subnets := range ranges {
+				if _, ipv6Subnet, err := net.ParseCIDR(subnets.Subnet); err == nil && strings.Contains(ipv6Subnet.String(), ":") {
+					totalIPv6, err := c.countTotalIPs(ipv6Subnet)
+					if err != nil {
+						glog.Errorf("Error calculating total IPs for IPv6 subnet %s: %v", ipv6Subnet.String(), err)
+						return err
+					}
+					assignedIPv6 += totalIPv6
+					c.assignedIPv6AddrCount = assignedIPv6
+				} else if _, ipv4Subnet, err := net.ParseCIDR(subnets.Subnet); err == nil {
+					totalIPv4, err := c.countTotalIPs(ipv4Subnet)
+					if err != nil {
+						glog.Errorf("Error calculating total IPs for IPv4 subnet %s: %v", ipv4Subnet.String(), err)
+						return err
+					}
+					assignedIPv4 += totalIPv4
+					c.assignedIPv4AddrCount = assignedIPv4
+				}
+			}
+		}
+	}
+	if assignedIPv4 > 0 {
```

Comment:
Discussed offline on this - let's start without this ratio first so that we are not unnecessarily consuming more CPU/memory for metrics exporting, given this is trivial to compute on the metric consumer end.

Reply:
Done